psse/psse-model/src/main/java/com/powsybl/psse/model/PsseRawReader.java

```diff
@@ -21,39 +23,255 @@
+        } catch (PsseException e) {
+            return false; // invalid PSS/E content
         }
-        return line.substring(0, slashIndex);
+
+        int ic = caseIdentification.getIc();
+        double sbase = caseIdentification.getSbase();
+        int rev = caseIdentification.getRev();
+        double basfrq = caseIdentification.getBasfrq();
+
+        if (ic == 0 && sbase > 0. && rev <= PsseConstants.SUPPORTED_VERSION && basfrq > 0.) {
+            return true;
+        }
+
+        return false;
     }
 
-    private static String readLineAndRemoveComment(BufferedReader reader) throws IOException {
-        String line = reader.readLine();
-        if (line == null) {
-            return null;
+    public PsseRawModel read(BufferedReader reader) throws IOException {
+        PsseContext context = new PsseContext();
+        return read(reader, context);
+    }
+
+    public PsseRawModel read(BufferedReader reader, PsseContext context) throws IOException {
+        Objects.requireNonNull(reader);
+        Objects.requireNonNull(context);
+
+        PsseCaseIdentification caseIdentification = readCaseIdentificationData(reader, context);
+        PsseRawModel model = new PsseRawModel(caseIdentification);
+
+        model.getBuses().addAll(readBusData(reader, context));
+        model.getLoads().addAll(readLoadData(reader, context));
+        model.getFixedShunts().addAll(readFixedBusShuntData(reader, context));
+        model.getGenerators().addAll(readGeneratorData(reader, context));
+        model.getNonTransformerBranches().addAll(readNonTransformerBranchData(reader, context));
+        model.getTransformers().addAll(readTransformerData(reader, context));
+        model.getAreas().addAll(readAreaInterchangeData(reader, context));
+
+        // 2-terminal DC data
+        readRecordBlock(reader); // TODO
+
+        // voltage source converter data
+        readRecordBlock(reader); // TODO
+
+        // impedance correction data
+        readRecordBlock(reader); // TODO
+
+        // multi-terminal DC data
+        readRecordBlock(reader); // TODO
+
+        // multi-section line data
+        readRecordBlock(reader); // TODO
+
+        model.getZones().addAll(readZoneData(reader, context));
+
+        // inter-area transfer data
+        readRecordBlock(reader); // TODO
+
+        model.getOwners().addAll(readOwnerData(reader, context));
+
+        // facts control device data
+        readRecordBlock(reader); // TODO
+
+        model.getSwitchedShunts().addAll(readSwitchedShuntData(reader, context));
+
+        // gne device data
+        readRecordBlock(reader); // TODO
+
+        // q record (nothing to do)
+        readRecordBlock(reader);
+
+        return model;
+    }
+
+    // Read blocks
+
+    private static PsseCaseIdentification readCaseIdentificationData(BufferedReader reader, PsseContext context) throws IOException {
+        String line = readLineAndRemoveComment(reader);
+        Objects.requireNonNull(line);
```

Comment:
```
String line = Objects.requireNonNull(readLineAndRemoveComment(reader));
```

Can the `readLineAndRemoveComment` really return null?

Same remark for all occurrences below

Reply:
Done. Now `readLineAndRemoveComment` never returns null